packages/animations/lib/src/fade_transition.dart

```diff
@@ -0,0 +1,232 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:animations/src/utils/curves.dart';
+import 'package:flutter/material.dart';
+
+/// Displays a modal above the current contents of the app.
+///
+/// This function displays the [FadeModalRoute], which transitions in
+/// with the Material fade transition.
+///
+/// Content below the modal is dimmed with a [ModalBarrier].
+///
+/// ```dart
+/// /// Sample widget that uses [showModalWithFadeTransition].
+/// class MyHomePage extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return Scaffold(
+///       body: Center(
+///         child: RaisedButton(
+///           onPressed: () {
+///             showModalWithFadeTransition(
+///               context: context,
+///               child: FlutterLogoModal(),
+///             );
+///           },
+///           child: Icon(Icons.add),
+///         ),
+///       ),
+///     );
+///   }
+/// }
+///
+/// /// Displays a modal with the FlutterLogo on it.
```

Comment:
This comment is kinda strange, nothing in this class seems to imply that it is a modal. It's only one because it was rendered with showModalWithFadeTransition, no?

Reply:
Right, that makes sense. I'll rename the widget